platform/drivers/display/display_libSDL.c

```diff
@@ -259,36 +116,39 @@ void display_terminate()
     {}         /* Wait until current render finishes */
     printf("Terminating SDL display emulator, goodbye!\n");
     free(frameBuffer);
-    SDL_DestroyWindow(window);
-    SDL_Quit();
 }
 
 void display_renderRows(uint8_t startRow, uint8_t endRow)
 {
     (void) startRow;
     (void) endRow;
-    PIXEL_SIZE *pixels;
-    int pitch = 0;
-    if (SDL_LockTexture(displayTexture, NULL, (void **) &pixels, &pitch) < 0)
+    inProgress = true;
+    if(!sdl_ready)
     {
-        printf("SDL_lock failed: %s\n", SDL_GetError());
+        sdl_ready = sdl_main_loop_ready();
```

Comment:
This whole `sdl_ready` local variable may be an over optimisation. It's just a way to avoid an extra mutex check on each loop.